app/models/case_contact_report.rb

```diff
@@ -2,11 +2,19 @@ class CaseContactReport
   attr_reader :case_contacts
 
   def initialize(args = {})
-    @case_contacts = if args[:start_date] && args[:end_date]
-      CaseContact.occurred_between(args[:start_date], args[:end_date])
-    else
-      CaseContact.all
+    @case_contacts = filtered_case_contacts(args)
+  end
+
+  def filtered_case_contacts(args)
+    return CaseContact.all if args.empty?
+    contact = CaseContact
```

Comment:
What's the benefit of doing this? Is it in case neither set of args that we are checking for exist? If so, we'd probably want to do `contact = CaseContact.all`.

Reply:
We are in the middle of adding 5 additional filters. If none of the 7 filters are present, then there's no reason to go through the query building process, and we should just return all CaseContacts.